async-nats/tests/jetstream_tests.rs

```diff
@@ -103,174 +101,145 @@ mod jetstream {
             .await
             .unwrap()
             .await
             .unwrap();
-        // 3 messages should be there, so this should error.
-        assert_eq!(
-            context
-                .send_publish(
-                    "foo".to_string(),
-                    Publish::build().expected_last_sequence(2),
-                )
-                .await
-                .unwrap()
-                .await
-                .unwrap_err()
-                .kind(),
-            PublishErrorKind::WrongLastSequence
-        );
-        // 3 messages there, should be ok for this subject too.
+
+        let info = stream.info().await.unwrap();
+        assert_eq!(3, info.state.messages);
+
         context
-            .send_publish(
-                "foo".to_string(),
-                Publish::build().expected_last_subject_sequence(3),
-            )
+            .publish("bar".to_string(), "data".into())
+            .expected_last_sequence(3)
             .await
             .unwrap()
             .await
             .unwrap();
-        // 4 messages there, should error.
-        assert_eq!(
-            context
-                .send_publish(
-                    "foo".to_string(),
-                    Publish::build().expected_last_subject_sequence(3),
-                )
-                .await
-                .unwrap()
-                .await
-                .unwrap_err()
-                .kind(),
-            PublishErrorKind::WrongLastSequence
-        );
 
-        // Check if it works for the other subjects in the stream.
+        let info = stream.info().await.unwrap();
+        assert_eq!(4, info.state.messages);
+
         context
-            .send_publish(
-                "bar".to_string(),
-                Publish::build().expected_last_subject_sequence(0),
-            )
+            .publish("foo".to_string(), "data".into())
+            .expected_last_subject_sequence(3)
             .await
             .unwrap()
             .await
             .unwrap();
-        // Sequence is now 1, so this should fail.
+
+        let info = stream.info().await.unwrap();
+        assert_eq!(5, info.state.messages);
+
         context
-            .send_publish(
-                "bar".to_string(),
-                Publish::build().expected_last_subject_sequence(0),
-            )
+            .publish("foo".to_string(), "data".into())
+            .expected_last_sequence(1)
             .await
             .unwrap()
             .await
             .unwrap_err();
-        // test header shorthand
-        assert_eq!(stream.info().await.unwrap().state.messages, 5);
-        context
-            .send_publish(
-                "foo".to_string(),
-                Publish::build().header(NATS_MESSAGE_ID, id.as_str()),
-            )
-            .await
-            .unwrap()
-            .await
-            .unwrap();
-        // above message should be ignored.
-        assert_eq!(stream.info().await.unwrap().state.messages, 5);
```

Comment:
I think this test is a victim of a rebase.
Let's rewrite it to what it should do.